volatility3/framework/layers/crash.py

```diff
@@ -67,34 +72,96 @@ def __init__(self, context: interfaces.context.ContextInterface, config_path: st
+                                   offset=0,
+                                   layer_name=self._base_layer)
+
+    def get_summary_header(self) -> interfaces.objects.ObjectInterface:
+        return self.context.object(self._crash_common_table_name + constants.BANG + "_SUMMARY_DUMP",
+                                   offset=0x1000 * self.headerpages,
+                                   layer_name=self._base_layer)
+
     def _load_segments(self) -> None:
         """Loads up the segments from the meta_layer."""
-        header = self.context.object(self._crash_table_name + constants.BANG + self.dump_header_name,
-                                     offset = 0,
-                                     layer_name = self._base_layer)
 
         segments = []
 
-        offset = self.headerpages
-        header.PhysicalMemoryBlockBuffer.Run.count = header.PhysicalMemoryBlockBuffer.NumberOfRuns
-        for x in header.PhysicalMemoryBlockBuffer.Run:
-            segments.append((x.BasePage * 0x1000, offset * 0x1000, x.PageCount * 0x1000, x.PageCount * 0x1000))
-            # print("Segments {:x} {:x} {:x}".format(x.BasePage * 0x1000,
-            #                  offset * 0x1000,
-            #                  x.PageCount * 0x1000))
-            offset += x.PageCount
+        if self.dump_type == 0x1:
+            header = self.context.object(self._crash_table_name + constants.BANG + self.dump_header_name,
+                                         offset=0,
+                                         layer_name=self._base_layer)
+
+            offset = self.headerpages
+            header.PhysicalMemoryBlockBuffer.Run.count = header.PhysicalMemoryBlockBuffer.NumberOfRuns
+            for x in header.PhysicalMemoryBlockBuffer.Run:
+                segments.append((x.BasePage * 0x1000, offset * 0x1000, x.PageCount * 0x1000, x.PageCount * 0x1000))
+                offset += x.PageCount
+
+        elif self.dump_type == 0x05:
+            summary_header = self.get_summary_header()
+            first_bit = None  # First bit in a run
+            first_offset = 0  # File offset of first bit
+            last_bit_seen = 0  # Most recent bit processed
+            offset = summary_header.HeaderSize  # Size of file headers
+            buffer_char = summary_header.get_buffer_char()
+            buffer_long = summary_header.get_buffer_long()
+
+            for outer_index in range(0, ((summary_header.BitmapSize + 31) // 32)):
+                if buffer_long[outer_index] == 0:
+                    if first_bit is not None:
+                        last_bit = ((outer_index - 1) * 32) + 31
+                        segment_length = (last_bit - first_bit + 1) * 0x1000
+                        segments.append((first_bit * 0x1000, first_offset, segment_length, segment_length))
+                        first_bit = None
+                elif buffer_long[outer_index] == 0xFFFFFFFF:
```

Comment:
Just a note, this looks like a constant based off the data type.  If it's unlikely to change type, that's ok, but if it's just defined as -1 then we probably ought to calculate what -1 would be...

Reply:
It shouldn't change type (unsigned long).